mailcap.c

```diff
@@ -32,7 +32,7 @@
 #include "mailcap.h"
 #include "xwrapper.h"
 
-#define DEFAULT_MAILCAP_ENTRY "*/*; "DEFAULT_OPENER" %s"
+#define DEFAULT_MAILCAP_ENTRY "*/*; "DEFAULT_OPENER" %s; needsterminal"
```

Comment:
and I guess this is needed, since we can't know in advance whether xdg-open/open will need a terminal or not.